qprofiler2/src/org/qcmg/qprofiler2/summarise/PositionSummary.java

```diff
@@ -152,7 +156,7 @@ public void addPosition(final int position, String  rgid  ) {
 		
 		rgCoverages[ readGroupIds.indexOf(rgid)  ].increment(position / BUCKET_SIZE);  
```

Comment:
what is the performance impact of looping through a list (`readGroupIds.indexOf(rgid)`) for each record?
If you don't want to use a map, would it be better to sort the array and call `binarySearch` instead?

Reply:
There were only a couple readGroups in one BAM, but now sometimes we received BAM with 30 read groups. O(log(N)) is much faster than O(N), especially we only sort the list once. 
OK, the code is updated